pkg/discovery/worker/dispatcher.go

```diff
@@ -184,17 +187,18 @@ func (d *SamplingDispatcher) ScheduleDispatch(nodes []*api.Node) {
 // Dispatch sampling discovery tasks. Each task to discover one node will be picked up by an available sampling discovery
 // worker. Set the timeout of finish assigning tasks of all nodes as given samplingInterval to avoid goroutine pile up.
 func (d *SamplingDispatcher) dispatchSamplingDiscoveries(nodes []*api.Node, samplingInterval time.Duration) {
-	finishCh := make(chan struct{})
-	stopCh := make(chan struct{})
-	defer close(finishCh)
-	defer close(stopCh)
+	// done channel indicates that a round of sampling is done
+	done := make(chan struct{})
+	// abort channel indicates that a round of sampling is aborted due to timeout
+	abort := make(chan struct{})
 	t := time.NewTimer(samplingInterval)
+	defer t.Stop()
 	// Dispatch tasks to the pool, which will be picked up by available sampling discovery workers
 	go func() {
 		for i, node := range nodes {
 			select {
-			case <-stopCh:
-				glog.Warningf("Dispatching sampling discovery tasks timeout. Collected data for %v out of %v nodes in this sampling cycle",
+			case <-abort:
+				glog.Warningf("Timeout dispatching sampling discovery tasks to %v out of %v nodes in this sampling cycle.",
```

Comment:
The original msg reads more appropriate, this one has become a little confusing.

Reply:
Fixed based on our discussion.